week3.js

```diff
@@ -0,0 +1,223 @@
+  	   	console.log("a " + color + " used car");
+  	   }
+  	   	else{console.log("a " + color + " car");}
+  	   }
+  else if(code === 2){
+  	   if(age > 0){
+  		console.log("a " + color + " used motorbike");
+  	   }
+  		else{
+  	   	console.log("a " + color + " motorbike");}
+  	   }
+}
+
+
+
+//7
+// method 1: using objects (adding another layer of objects for fun!)
+vehicle = {
+	"motor": {
+		id: "motorbike",
+    	id2: "bus",
+     	id3: "truck",
+     	id4: "car"
+     },
+     "railed": {
+     	id: "train",
+     	id2: "tram",
+     },
+     "aircraft": {
+     	id: "airplane",
+     	id2: "helicopter",
+     }
+}
+const firstKey = Object.keys(vehicle)[0];
+const secondKey = Object.keys(vehicle)[1];
+const thirdKey = Object.keys(vehicle)[2];
+//----------------------------------------------------
+console.log(vehicle["aircraft"]["id2"]);
+//logs helicopter
+console.log(vehicle[thirdKey]["id2"]);
+//logs helicopter
+//----------------------------------------------------
+
+//method 2: using arrays
+const vehicles = ["motorbike", "caravan", "bike", "bus"];
+
+
+
+
+//8
+console.log(vehicles[2]);
+
+
+
+
+//9
+const vehicles = ["motorbike", "caravan", "bike", "bus"];
+
+
+function vehicle(color,code,age){
+  let i;
+  if(code === 1){
+  	i = vehicles[0];
+  }
+  else if(code === 2){
+  	i = vehicles[1];
+  }
+  else if(code === 3){
+  	i = vehicles[2]
+  } 
+  else if(code === 4){
+  	i = vehicles[3]
+  } 
+
+  let x;
+  if(age <= 1){
+  x = "new";
+  }
+  else{
+	x = "used";
+  }
+   console.log("a",color,x,i);
+}
+vehicle("green", 3, 1)
+
+/////////////////////////////////////////////////////////
+//10
+const vehicles = ["car", "motorbike", "caravan", "bike"];
+
+let vehicleToService="";
+for(let counter = 0; counter < vehicles.length; counter ++){
+  if(counter === vehicles.length - 1){
+  	vehicleToService += vehicles[counter] + "s.";
+  } 
+  else {vehicleToService += vehicles[counter] + "s, ";}
+ }
+
+ console.log(" \"Amazing Joe's Garage, we service" , vehicleToService,"\"");
+
+
+
+//11
+const vehicles = ["car", "motorbike", "caravan", "bike"];
+vehicles.push("plane");
+console.log(vehicles);
+let vehicleToService="";
+for(let counter = 0; counter < vehicles.length; counter ++){
+  if(counter === vehicles.length - 1){
+  	vehicleToService += vehicles[counter] + "s.";
+  } 
+  else {vehicleToService += vehicles[counter] + "s, ";}
+ }
+
+console.log(" \"Amazing Joe's Garage, we service" , vehicleToService,"\"");
+
+
+
+//12
+const object = {};
+
+
+
+//13
+const person = {htmlCss:"Philipp Beau", cli:"Unmesh Joshi", JavaScript:"Yash Kapila"};
+
+
+
+//16
+let o1 = { foo: "bar" };
+let o2 = { foo: "bar" };
+let o3 = o2;
+
+console.log(o3);
+o2.foo = "something";
+console.log(o3);
+o1.foo = "and something else";
+console.log(o3);
+
+//Does the order that you assign (o3 = o2 or o2 = o3) matter?
+//yes, because the right side is being assigned to the left side, otherwise it will show that the variable is not defined.
```

Comment:
Yes that is correct. You changed `foo` on the `o2` object, try this exercise again but this time change the whole object so `o2 = { foo: "something" }`. See if that changes anything.